spec/safe_statements_spec.rb

```diff
@@ -2518,6 +2518,49 @@ def up
               test_migration.suppress_messages { test_migration.migrate(:up) }
             end.to raise_error(PgHaMigrations::InvalidMigrationError, "Concurrent partitioned index creation not supported on Postgres databases before version 11")
           end
+
+          it "creates valid index with nulls_not_distinct when multiple child partitions exist" do
+            create_range_partitioned_table(:foos3, migration_klass, with_partman: true)
+
+            test_migration = Class.new(migration_klass) do
+              def up
+                safe_add_concurrent_partitioned_index(
+                  :foos3,
+                  :text_column,
+                  nulls_not_distinct: true
+                )
+              end
+            end
+
+            # Check that we pass the nulls_not_distinct option to the underlying add_index method
+            expect(ActiveRecord::Base.connection).to receive(:add_index).with(
+              anything,
+              anything,
+              hash_including(nulls_not_distinct: true)
+            ).at_least(:once).and_call_original
```

Comment:
I think we would know the exact number of times this should be called and the expectations should be more precise